corgi/tasks/brew.py

```diff
@@ -494,19 +505,31 @@ def fetch_unprocessed_relations(
 ) -> int:
     relations_query = (
         ProductComponentRelation.objects.filter(type=relation_type, created_at__gte=created_since)
-        .values_list("build_id", flat=True)
+        .values_list("build_id", "build_type")
         .distinct()
         .using("read_only")
     )
     logger.info(f"Processing relations of type {relation_type}")
     processed_builds = 0
-    for build_id in relations_query.iterator():
+    for build_id, b_type in relations_query.iterator():
         if not build_id:
             # build_id defaults to "" and int() will fail in this case
             continue
-        if not SoftwareBuild.objects.filter(build_id=int(build_id)).using("read_only").exists():
+        if (
+            not SoftwareBuild.objects.filter(build_id=int(build_id), build_type=b_type)
+            .using("read_only")
+            .exists()
+        ):
             logger.info("Processing CDN relation build with id: %s", build_id)
-            slow_fetch_modular_build.delay(build_id, force_process=force_process)
+            if b_type == SoftwareBuild.Type.CENTOS:
+                # This skips use of the Collector models for builds in the CENTOS koji instance
+                # It was done to avoid updating the collector models not to use build_id as
+                # a primary key. It's possible because the only product stream (openstack-rdo)
+                # stored in CENTOS koji doesn't use modules
+                slow_fetch_brew_build.delay(
+                    build_id, SoftwareBuild.Type.CENTOS, force_process=force_process
+                )
```

Comment:
This makes sense, but are we missing some logic at the end? Right now we call slow_fetch_brew_build and slow_fetch_modular_build, when I think we're only supposed to call one:
```suggestion
                slow_fetch_brew_build.delay(
                    build_id, SoftwareBuild.Type.CENTOS, force_process=force_process
                )
                processed_builds += 1
                continue
```

Reply:
Good catch again. I'll update it using your suggestion.